lahja/endpoint.py

```diff
@@ -145,8 +145,9 @@ def _process_item(self, item: BaseEvent, config: BroadcastConfig) -> None:
 
         if in_futures:
             future = self._futures[config.filter_event_id]
-            future.set_result(item)
-            self._futures.pop(config.filter_event_id)
+            if not future.cancelled() and not future.done():
```

Comment:
The check is redundant: `done` includes `cancelled`: https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.done

Reply:
Ah, right!